src/main/java/io/github/tryexceptelse/jdex/JDex.java

```diff
@@ -104,38 +237,59 @@ public void setUi(Ui ui) {
 
   /**
    * Gets currently active rolodex object
-   * @return IDex: Active IDex object
+   *
+   * @return IRolodex: Active IRolodex object
    */
-  public Dex getDex() {
-    return dex;
+  public IRolodex getRolodex() {
+    return rolodex;
   }
 
   /**
-   * Sets active rolodex object to new IDex
-   * @param dex: New Rolodex object to be used
+   * Sets active rolodex object to new IRolodex
+   *
+   * @param rolodex: New Rolodex object to be used
    */
-  public void setDex(Dex dex) {
-    this.dex = dex;
+  public void setRolodex(IRolodex rolodex) {
+    this.rolodex = rolodex;
+  }
+
+  /**
+   * gets runtime args main method was called with
+   * @return runtime args String[] array
+   */
+  public String[] getRuntimeArgs() {
+    return runtimeArgs;
   }
 
   /**
    * builds rolodex object with information from settings
-   * @return IDex: Created rolodex object
+   *
+   * @return IRolodex: Created rolodex object
    */
-  private Dex buildDex() {
+  private IRolodex buildRolodex() {
     // placeholder: should check settings for the last save,
     // and load that if possible. If none exists, or error is raised,
     // construct new
-    return new Dex();
+    return new Rolodex(contactsFile);
   }
 
   /**
    * builds user interface based on information in settings
+   *
    * @return Ui: User Interface object
    */
-  private Ui buildUi(){
-    return new Gui(this, runtimeArgs); //placeholder.
-      // Should first check settings for type of GUI
+  private Ui buildUi() {
+    return new Gui(this); //placeholder.
+    // Should first check settings for type of GUI
   }
 
-}
+  /**
+   * creates contacts File object to be stored and
+   * used for saving and loading data
+   * @return File to save to and load from.
+     */
+  private File buildContactsFile() {
+      // File does not accept a default constructor call.
```

Comment:
its methods presume it represents an actual file location that can be accessed and read from/written to